src/Integration.UnitTests/Suppression/SonarQubeIssuesProviderTests.cs

```diff
@@ -205,95 +205,250 @@ public void GetIssues_NoIssuesOnServer_ReturnsEmptyList()
-
-            SetupSolutionBinding(isConnected: true,
-                issues: new List<SonarQubeIssue> { issue1, issue2, issue3, issue4, issue5, issue6, issue7, issue8, issue9, issue1 });
-
-            var issuesProvider = new SonarQubeIssuesProvider(mockSqService.Object, "sqKey",
-                mockTimerFactory.Object, testLogger);
+
+            SetupSolutionBinding(true, new List<SonarQubeIssue> { sonarQubeIssue });
+
+            var issuesProvider = new SonarQubeIssuesProvider(mockSqService.Object, "sqkey", mockTimerFactory.Object, testLogger);
             WaitForInitialFetchTaskToStart();
 
             VerifyServiceGetIssues(Times.Exactly(1)); // issues should be fetched on creation
 
-            // 1. Project id doesn't match -> no issues
-            var matches = issuesProvider.GetSuppressedIssues("unrecognisedProjectId", "folder1/file1");
-            matches.Should().BeEmpty();
-            VerifyServiceGetIssues(Times.Exactly(1), "sqKey"); // cached issues should be used
+            // Act
+            var matches = issuesProvider.GetSuppressedIssues("projectId", null);
 
-            // 2. File id doesn't match -> no issues
-            matches = issuesProvider.GetSuppressedIssues("projectID1", "folder1/filexxx");
-            matches.Should().BeEmpty();
+            // Assert
+            matches.Should().HaveCount(1);
+        }
 
-            // 3. File id and guid match -> issues returned
-            matches = issuesProvider.GetSuppressedIssues("projectID1", "file1");
-            matches.Count().Should().Be(1);
-            CheckExpectedIssueReturned("hash8", matches);
+        [TestMethod]
+        public void GetSuppressedIssues_WhenProjectHasNoModules_WhenIssueIsModuleLevelAndFound_ReturnsExpectedIssue()
+        {
+            // Arrange
+            var sonarQubeIssue = new SonarQubeIssue(null, null, null, "message", "sqkey",
+                SonarQubeIssueResolutionState.FalsePositive, "S101");
 
-            // 4. File id and guid match, case-insensitive -> issues returned
-            matches = issuesProvider.GetSuppressedIssues("PROJECTID2", "FOLDER1/FILE1");
-            matches.Count().Should().Be(2);
-            CheckExpectedIssueReturned("hash2", matches);
-            CheckExpectedIssueReturned("hash3", matches);
+            SetupSolutionBinding(true, new List<SonarQubeIssue> { sonarQubeIssue },
+                new List<SonarQubeModule> { new SonarQubeModule("sqkey", "", "") });
 
-            VerifyServiceGetIssues(Times.Exactly(1));
+            var issuesProvider = new SonarQubeIssuesProvider(mockSqService.Object, "sqkey", mockTimerFactory.Object, testLogger);
+            WaitForInitialFetchTaskToStart();
+
+            VerifyServiceGetIssues(Times.Exactly(1)); // issues should be fetched on creation
+
+            // Act
+            var matches = issuesProvider.GetSuppressedIssues("projectId", null);
+
+            // Assert
+            matches.Should().HaveCount(1);
```

Comment:
Same: the mocked list of issues should contain an issue that should not be returned.